File: CTask/src/org/ShapeSHifter420/Fabric/GenderPushFactory.java
package org.ShapeSHifter420.Fabric;

import org.ShapeSHifter420.Pushes.GenderPush;
import org.ShapeSHifter420.Pushes.IPush;

import java.util.Map;

public class GenderPushFactory implements IPushFactory{
    @Override
    public IPush create(Map<String, String> conf) {
        String text = conf.getOrDefault("text", "");
        String gender = conf.getOrDefault("gender", "none");
        return new GenderPush(text,gender);
    }
}

File: CKotlin/Factory/GenderPushFactory.kt
package Factory

import Pushes.GenderPush
import Pushes.IPush

object GenderPushFactory : IPushFactory {
    override fun made(param: Map<String, String>): IPush = GenderPush(
        gender = param["gender"] ?: "none",
        text = param["text"] ?: "none"
    )
}